src/stats.rs

```diff
@@ -502,6 +609,7 @@ impl Collector {
                     // Update latest stats used in SHOW STATS
                     let mut guard = LATEST_STATS.lock();
                     for (key, value) in stats.iter() {
+                        self.statsd_client.send_gauge(key, value.clone() as u64);
```

Comment:
Issuing network calls while holding a mutex isn't a good idea.

Reply:
Nice call out, will move it outside into it's own loop